server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java

```diff
@@ -479,6 +479,48 @@ public void testSimpleMailSendWithEHLO() throws Exception {
     }
 
     @Test
+    public void messageExceedingMessageSizeShouldBeDiscarded() throws Exception {
+        init(smtpConfiguration);
+        int maxSize = 1024;
+        smtpServer.setMaximalMessageSize(maxSize);
+
+        SMTPClient smtpProtocol = new SMTPClient();
+        InetSocketAddress bindedAddress = new ProtocolServerUtils(smtpServer).retrieveBindedAddress();
+        smtpProtocol.connect(bindedAddress.getAddress().getHostAddress(), bindedAddress.getPort());
+
+        // no message there, yet
+        assertThat(queue.getLastMail())
```

Comment:
It's not new that a good test is prepare/run/assert.
You are mixing assertions, so you either need another test for that (no matter what it means) or you doesn't need this assert.
Usually, the assert should be similar to the test title.